router/http/src/main/scala/io/buoyant/router/Http.scala

```diff
@@ -32,6 +32,7 @@ object Http extends Router[Request, Response] with FinagleServer[Request, Respon
     val client: StackClient[Request, Response] = FinagleHttp.client
       .transformed(StackRouter.Client.mkStack(_))
       .transformed(_.replace(http.TracingFilter.role, http.TracingFilter.module))
+      .transformed(_.insertBefore(http.TracingFilter.role, DiscardProxySideEffects.module))
```

Comment:
Finally, I think that this needs to be installed on the serverside and not the client-side.

We want this rewriting to happen before we apply an Identifier to the request (which happens above the client).